### PR TITLE
Automated Tests for Actions/Reducers/Components

### PR DESCRIPTION
**Pull request by Chiu-Kevin-bah**
- Added automated unit tests for `./src/actions`
- Added automated unit tests for `./src/components`
- Added automated unit tests for `./src/reducers`

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,10 +38,25 @@
     "eslint-plugin-react": "^7.10.0",
     "sinon": "^6.1.3"
   },
+  "devDependencies": {
```
Comment (mjoyce91) [inline package.json]: Combine with other `devDependencies` object in package.json

File: src/Components/Details.js [hunk 2]
```diff
@@ -21,4 +21,4 @@ const Details = ({ data, id }) => (
   </Container>
 );
 
-export default Details;
+export default DetailsComponent;
```
Comment (mjoyce91) [inline src/Components/Details.js]: `Details` to match declaration

File: package.json (hunk 1 above)
Comment (mjoyce91) [reply package.json]: Also I don't think `yarn install` is picking these up.

File: src/tests.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// can be exported from the Container/Component. The second is a config
+// object with any parameters we want to pass to the function with the same name.
+// For example, to pass the argument "test" to the getData() function, and "test"
+// and {1} to submitData() function, we could create an object such as
+// { getData: ["test"], submitData: ["test", 1] }
+// Then we spy dispatch and call each property of the mapDispatchToProps
+// object. Calling each property should trigger the call count on the spy,
+// which we test. We reset call count between tests since we're using the
+// calledOnce method.
+export function testDispatchFunctions(mapDispatchToProps, config = {}) {
```
Comment (mjoyce91) [inline src/tests.js:18]: This looks familiar :grinning:

**Review by mjoyce91 — DISMISSED** [review]
See comments

File: src/Components/index.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import * as glob from 'glob';
+import * as path from 'path';
+
+const options = { cwd: __dirname, ignore: './*.test.js' };
+const files = glob.sync('./*.js', options);
+
+describe('Components', () => {
+  let i;
+
+  for(i = 0; i < files.length; i++) {
```
Comment (mjoyce91) [inline src/Components/index.test.js]: `files.forEach`?

File: src/tests.js (hunk 3 above)
Comment (Chiu-Kevin-bah) [reply src/tests.js:18]: Why reinvent the wheel? 😉

File: src/Components/index.test.js (hunk 4 above)
Comment (Chiu-Kevin-bah) [reply src/Components/index.test.js]: Yea I always had `forEach` async issues....I dunno why...but I don't with `map`....is that weird?

At least when it's on the server-side. I don't actually remember async issues on client side

File: src/Components/Details.js (hunk 2 above)
Comment (Chiu-Kevin-bah) [reply src/Components/Details.js]: Good call! I fixed the `HomePage` one too

File: src/Components/index.test.js (hunk 4 above)
Comment (Chiu-Kevin-bah) [reply src/Components/index.test.js]: Maybe it's related to this?
https://coderwall.com/p/kvzbpa/don-t-use-array-foreach-use-for-instead

I started with a `forEach` and the test finished complaining about having no tests.

Being slower? I'm sure if this is true, u can't tell most of the time. Most of the time when I experience this it isn't a small array either so shrug.

File: src/api.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios';
 
 export const config = {
-  baseURL: localStorage.getItem('API_URL') || 'http://jsonplaceholder.typicode.com',
+  baseURL: localStorage.getItem('API_URL'),
 };
```
Comment (Chiu-Kevin-bah) [inline src/api.js:5]: Just wanted to note this change: Due to what ended up doing in the entry file for verifying `api.config`, `API_URL` wouldn't ever be undefined unless `api.config` had a bad value => In which case well I dunno lol. 

I think this eliminates the tracing baggage ^^ lol

**Pull request merged**